pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -603,17 +613,17 @@ private PGStream enableSSL(PGStream pgStream, SslMode sslMode, Properties info,
     }
   }
 
-  private void sendStartupPacket(PGStream pgStream, List<String[]> params)
+  private void sendStartupPacket(PGStream pgStream, List<StartupParam> params)
       throws IOException {
     if (LOGGER.isLoggable(Level.FINEST)) {
       StringBuilder details = new StringBuilder();
       for (int i = 0; i < params.size(); ++i) {
         if (i != 0) {
           details.append(", ");
         }
-        details.append(params.get(i)[0]);
+        details.append(params.get(i).key);
         details.append("=");
-        details.append(params.get(i)[1]);
+        details.append(params.get(i).value);
       }
```

Comment:
Since you now have a class here, write a toString() to output "key=value"

Reply:
That make sense, I've applied that suggestion and also added `getEncodedKey()` and `getEncodedValue()`